native/src/routes/feedback/containers/FeedbackModalContainer.js

```diff
@@ -1,33 +1,296 @@
+
+const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsType => {
+  const refreshProps = {
+    navigation: ownProps.navigation
+  }
+  if (state.cities.status === 'error') {
+    return { status: 'error', message: state.cities.message, code: state.cities.code, refreshProps }
+  }
+
+  if (state.cities.status === 'loading') {
+    return { status: 'loading', progress: 0 }
+  }
+  return {
+    status: 'success',
+    innerProps: {
+      cities: state.cities.models,
+      navigation: ownProps.navigation,
+      t: ownProps.t
+    },
+    refreshProps
+  }
+}
+
+const mapDispatchToProps = (dispatch: Dispatch<StoreActionType>): DispatchPropsType => ({ dispatch })
+
+const refresh = (refreshProps: RefreshPropsType, dispatch: Dispatch<StoreActionType>) => {
+  const { navigation } = refreshProps
+  const feedbackInformation = navigation.state.params
+  const navigateToFeedback = createNavigateToFeedbackModal(navigation)
+  navigateToFeedback(feedbackInformation)
+}
+
+export type SendingStatusType = 'idle' | 'sending' | 'failed' | 'successful'
+
+type FeedbackModalStateType = {|
+  feedbackOptions: Array<FeedbackVariant>,
+  selectedFeedbackIndex: number,
+  comment: string,
+  sendingStatus: SendingStatusType
+|}
+
+class FeedbackModalContainer extends React.Component<ContainerPropsType, FeedbackModalStateType> {
+  constructor (props: ContainerPropsType) {
+    super(props)
+    const feedbackOptions = this.getFeedbackOptions()
+    this.state = { feedbackOptions, selectedFeedbackIndex: 0, comment: '', sendingStatus: 'idle' }
+  }
+
+  getCityName = (): string => {
+    const { cities, navigation } = this.props
+    const cityCode = navigation.getParam('cityCode')
+    return cityCode ? CityModel.findCityName(cities, cityCode) : INTEGREAT_INSTANCE
+  }
+
+  getFeedbackOptions = (): Array<FeedbackVariant> => {
+    const options = []
+    const currentPageFeedbackOption = this.getCurrentPageFeedbackOption()
+    if (currentPageFeedbackOption) {
+      options.push(currentPageFeedbackOption)
+    }
+
+    const contentFeedbackOption = this.getContentFeedbackOption()
+    if (contentFeedbackOption) {
+      options.push(contentFeedbackOption)
+    }
+
+    this.getOffersFeedbackOptions().forEach(option => options.push(option))
+    options.push(this.getTechnicalFeedbackVariant())
+
+    return options
+  }
+
+  getTechnicalFeedbackVariant = () => {
+    return new FeedbackVariant({
+      label: this.props.t('technicalTopics'),
+      feedbackType: CATEGORIES_FEEDBACK_TYPE,
+      feedbackCategory: TECHNICAL_FEEDBACK_CATEGORY
+    })
+  }
+
+  getContentFeedbackOption = (): ?FeedbackVariant => {
+    const { navigation, t } = this.props
+    const cityName = this.getCityName()
+
+    if (cityName) {
+      const label = t('contentOfCity', { city: cityName })
+      const feedbackCategory = CONTENT_FEEDBACK_CATEGORY
+      const feedbackType = navigation.getParam('type')
+      switch (feedbackType) {
```

Comment:
~~I think if you pass the right feedback type directly you could get rid of this, don't you?~~
Probably I am wrong. However, I think you have duplicated `EVENTS` feedback if you are at the events route. Not sure about that either though.

Reply:
Tested. There is only on Event feedback.